v2/workloadapi/bundlesource.go

```diff
@@ -98,31 +133,22 @@ func (s *BundleSource) setX509Context(x509Context *X509Context) {
 
 	newBundles := x509Context.Bundles.Bundles()
 
-	// Add/replace the X.509 content in the SPIFFE bundles. Track the trust
-	// domains represented in the new X.509 context so we can determine if
-	// which existing bundles need their X.509 content removed.
+	// Add/replace the X.509 authorities from the X.509 context. Track the trust
+	// domains represented in the new X.509 context so we can determine which
+	// existing trust domains are no longer represented.
 	trustDomains := make(map[spiffeid.TrustDomain]struct{}, len(newBundles))
 	for _, newBundle := range newBundles {
 		trustDomains[newBundle.TrustDomain()] = struct{}{}
-		existingBundle, ok := s.bundles.Get(newBundle.TrustDomain())
-		if !ok {
-			s.bundles.Add(spiffebundle.FromX509Bundle(newBundle))
-			continue
-		}
-		existingBundle.SetX509Authorities(newBundle.X509Authorities())
+		s.x509Authorities[newBundle.TrustDomain()] = newBundle.X509Authorities()
```

Comment:
This should be protected by a lock.

Reply:
It is. See the top of the setX509Context function.

Reply:
Sorry, I didn't expand those lines in the GitHub UI 🤦